src/resolve/resolved-conf.c

```diff
@@ -490,6 +490,56 @@ int config_parse_dns_stub_listener_extra(
         return 0;
 }
 
+int config_parse_resolve_restrict_interfaces(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Manager *m = userdata;
+        const char *p;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                m->restrict_interfaces = set_free(m->restrict_interfaces);
+                return 0;
+        }
+
+        for (p = rvalue;;) {
+                _cleanup_free_ char *n = NULL;
+
+                r = extract_first_word(&p, &n, NULL, 0);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to parse of restrict interfaces, ignoring assignment: %s",
+                                   rvalue);
+                        return 0;
+                }
+                if (r == 0)
+                        return 0;
+
+                r = set_ensure_put(&m->restrict_interfaces, NULL, n);
+                if (r < 0)
+                        log_syntax(unit, LOG_ERR, filename, line, r,
+                                   "Failed to store interface '%s' to restrict list, ignoring assignment: %m", n);
+
+                TAKE_PTR(n);
+        }
+
+        return 0;
```

Comment:
As I commented before, this is not necessary, as the above `for` loop does not break. Please drop this unnecessary `return`.